easyrest/views/restaurant_controler.py

```diff
@@ -123,3 +123,48 @@ def get_restaurant_controler(request):
         rest_with_tags = asign_tags([query])
         body = wrap([rest_with_tags])
     return body
+
+
+@view_config(route_name='get_my_restaurants', renderer='json', request_method='GET')
+def get_my_restaurant_controler(request):
+    """GET request controler to return my restaurants and
+    its tags
+    Args:
+        request: current pyramid request
+    Returns:
+        Json string(not pretty) created from dictionary with format:
+            {
+                "data": data,
+                "success": success,
+                "error": error
+            }
+        Where data is list with restaurant with tags assigned.
+        Style:
+            [
+                {
+                    "id": "restaurantId" + id,
+                    "name": (str),
+                    "description": (str),
+                    "addres_id": curently str,
+                    "owner_id": (int),
+                    "menu_id": (int)
+                    "tags": [{
+                        "id": "tagId" + id
+                        "name": tag name
+                        "priority": (int)
+                    }, ]
+                },
+            ]
+    """
+    owner = request.headers.get('Authorization')
+    # TODO: take owner_id from table users after Max pullrequst
+    # TODO: token = request.headers.get('Authorization')
+    # TODO: own = request.dbsession.query(User).filter_by(token=token).first()
+    # TODO: owner = own[0]['owner_id']
+    rests = asign_tags(request.dbsession.query(Restaurant).filter_by(owner_id=owner).all())
+    if not rests:
+        raise HTTPNotFound("No restaurants in database")
```

Comment:
In our project, the "user" becomes the "owner" when adding a restaurant, then the "owner" must have a restaurant. Now there is no function to remove the restaurant, but when it will be, the "owner" will become "user" if there is no restaurant.

Reply:
Users will not be able to view their restaurants, the button "show my restaurants" is available only for the owner... But now I understand, that owner by default will have restaurants and we do not need this verification... Is it better to delete it?

Reply:
Yep, remove this check and return empty list.

My point is next, from API position there is no any error: frontend asks for restaurants and it returns how much as possible. If it is an error from UI point of you, you may handle it on fronted's side, and render some error message, but not on backed side.

Reply:
I understand you,  if there are no restaurants, then this is the owner's problem:) backend did its job right(return restaurants)
Thanks.
I fixed it